src/PermitTokens/permitSignature.ts

```diff
@@ -0,0 +1,131 @@
+import { Wallet } from "ethers";
+
+async function getCorrectPermitSig(
+    wallet: Wallet,
+    token: any,
+    spender: string,
+    value: any,
+    deadline: any,
+    optional?: { nonce?: number; name?: string; chainId?: number; version?: string }
+    ) { 
+    const [nonce, name, version, chainId] = await Promise.all([
+        optional?.nonce ?? token.nonces(wallet.address),
+        optional?.name ?? token.name(),
+        optional?.version ?? '1',
+        optional?.chainId ?? wallet.getChainId(),
```

Comment:
we can use the sdk to create a multicall here (we only do a single network request so executes quicker)

Reply:
when do i do a multicall for the signatures, it oddly takes 2x as long..not sure why